relecov_tools/mail.py

```diff
@@ -1,67 +1,123 @@
-"""
-=============================================================
-HEADER
-=============================================================
-INSTITUTION: BU-ISCIII
-AUTHOR: Guillermo J. Gorines Cordero
-MAIL: guillermo.gorines@urjc.es
-VERSION: 0
-CREATED: 7-3-2022
-REVISED: 7-3-2022
-REVISED BY: guillermo.gorines@urjc.es
-DESCRIPTION:
+import yaml
+import json
+import logging
+import os
+import smtplib
+import relecov_tools.utils
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from email.mime.base import MIMEBase
+from email import encoders
+from jinja2 import Environment, FileSystemLoader
 
-    Includes the Email, and its associated methods.
 
-REQUIREMENTS:
-    -Python
+log = logging.getLogger(__name__)
 
-TO DO:
 
+class EmailSender:
+    def __init__(self, validate_file, config):
+        self.validate_file = validate_file
+        self.config = config
+        self.validate_data = relecov_tools.utils.read_json_file(validate_file)
+        self.template_path = self.config.get("delivery_template_path_file")
+        self.yaml_cred_path = self.config.get("yaml_cred_path")
 
-================================================================
-END_OF_HEADER
-================================================================
-"""
+        if not self.config:
+            raise ValueError("Configuration not loaded correctly.")
 
-# Imports
-import smtplib
+        if not self.validate_data:
+            raise ValueError("Validation data is not available.")
 
-from email.mime.multipart import MIMEMultipart
-from email.mime.text import MIMEText
+        if not os.path.exists(self.template_path):
+            raise FileNotFoundError(
+                f"The template file could not be found in path {self.template_path}.")
+
+    def get_invalid_count(self):
+        invalid_count = 0
+
+        for entry_key, entry_value in self.validate_data.items():
+            if "samples" in entry_value:
+                samples = entry_value["samples"]
+                for sample_key, sample_value in samples.items():
+                    if "valid" in sample_value and not sample_value["valid"]:
+                        invalid_count += 1
+        return invalid_count
```

Comment:
Move to `log_summary.py`
```suggestion
    def get_invalid_count(validation_logs):
        invalid_count = 0

        for entry_key, entry_value in validation_logsitems():
            if "samples" in entry_value:
                samples = entry_value["samples"]
                for sample_key, sample_value in samples.items():
                    if "valid" in sample_value and not sample_value["valid"]:
                        invalid_count += 1
        return invalid_count
```